### PR TITLE
Feature/209 configuration setup

### PR DESCRIPTION
**Pull request by KzuDemEvin**
# Issue #209 
* Move environment variables from frontend to backend
* Add REST-Controller to recieve config required for the frontend
* Disable jira services if not all required parameters are set
* Update/Fix docs

### CODE REVIEW

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 1]
```diff
@@ -50,6 +50,10 @@ public class ProjectManagementController {
   @GetMapping(value = "/oauth1/requestToken")
   public ResponseEntity<JiraRequestToken> getRequestToken() {
     LOGGER.debug("--> getRequestToken()");
+    if (!jiraServerService.serviceEnabled()) {
+      LOGGER.warn("Jira Server is not configured!");
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:55]: Please add also the error message to the exception.

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 2]
```diff
@@ -60,6 +64,10 @@ public ResponseEntity<JiraRequestToken> getRequestToken() {
   public ResponseEntity<TokenIdentifier> getOauth1AccessToken(
       @RequestBody VerificationCode verificationCode) {
     LOGGER.debug("--> getOauth1AccessToken()");
+    if (!jiraServerService.serviceEnabled()) {
+      LOGGER.warn("Jira Server is not configured!");
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:69]: Please add also the error message to the exception.

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 3]
```diff
@@ -73,6 +81,10 @@ public ResponseEntity<TokenIdentifier> getOauth1AccessToken(
   public ResponseEntity<TokenIdentifier> getOAuth2AccessToken(
       @RequestHeader("Origin") String origin, @RequestBody VerificationCode authorizationCode) {
     LOGGER.debug("--> getOAuth2AccessToken(), origin={}", origin);
+    if (!jiraCloudService.serviceEnabled()) {
+      LOGGER.warn("Jira Cloud is not configured!");
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:86]: Please add also the error message to the exception.

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 4]
```diff
@@ -90,6 +102,9 @@ public ResponseEntity<List<Project>> getProjects(
     if (projectManagementProvider == null) {
       LOGGER.warn("Bad Request: projectManagementProvider is null!");
       response = new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+    } else if (!projectManagementProvider.serviceEnabled()) {
+      LOGGER.warn("projectManagementProvider is not configured!");
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:107]: Please add also the error message to the exception.

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 5]
```diff
@@ -109,6 +124,9 @@ public ResponseEntity<List<UserStory>> getIssues(
...
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:129]: Please add also the error message to the exception.

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 6]
```diff
@@ -145,6 +163,9 @@ public ResponseEntity<String> createIssue(
     if (projectManagementProvider == null) {
       LOGGER.error("Failed to create issue!");
       throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Failed to create issue");
+    } else if (!projectManagementProvider.serviceEnabled()) {
+      LOGGER.warn("projectManagementProvider is not configured!");
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:168]: Please add also the error message to the exception.

File: backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java [hunk 7]
```diff
@@ -163,6 +184,9 @@ public void deleteIssue(
 
     if (projectManagementProvider == null) {
       LOGGER.error("Could not delete issue!");
+    } else if (!projectManagementProvider.serviceEnabled()) {
+      LOGGER.warn("projectManagementProvider is not configured!");
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (stritti) [inline backend/src/main/java/io/diveni/backend/controller/ProjectManagementController.java:189]: Please add also the error message to the exception.

**Review by stritti — CHANGES_REQUESTED** [review]
Please add error message also to exception to expose to client too.
Could we add a link to docs in the exception text to give a hint how to solve the issue?

**Comment by KzuDemEvin** [comment]
> Could we add a link to docs in the exception text to give a hint how to solve the issue?

I think this would be problematic in the case the documentation changes. You have to keep track of all the references.
A workaround would be to work with error codes.

**Comment by KzuDemEvin** [comment]
# Visualized changes in the docs: (including collapsible at the bottom)
image

## 1. unfolded:
image

## 2. unfolded:
image

**Review by SponsoredByPuma — APPROVED** [review]

**Pull request merged**